test/test_module_init.py

```diff
@@ -0,0 +1,369 @@
+import inspect
+import torch
+import unittest
+from unittest.mock import MagicMock, patch
+from torch.testing._internal.common_utils import TestCase, run_tests
+from torch.testing._internal.common_device_type import instantiate_device_type_tests
+
+
+# Returns a database of args & kwargs that can be used to construct each module.
+# Each entry is in class -> (args, kwargs) format.
+# Example: torch.nn.Linear -> ([10, 5], {})
+def build_constructor_arg_db():
```

Comment:
Alternatively, there's a similar DB in `common_nn.py` or something. It didn't have constructor args for every module, and in some cases, I need to pick different args to ensure paths are taken that create params / buffers.

Whenever `ModuleInfo` (analogous to the new `OpInfo`) comes about, this should be dealt with there.

Reply:
+1 for ModuleInfo (just imagine if there was already a basic version!)

Why is this a function that returns a dict instead of just being a dict?

Reply:
The function is left over from when I started with the `common_nn.py` `new_module_tests` and only filled in entries for the missing modules. Could make it just a dict but I kinda liked the mildly functional style